### PR TITLE
Assert the Asmth with exact kernel.

### PR DESCRIPTION
**Pull request by rainwoodman**
Also fixes a problem with NTAB -- it was picking undefined values.

### CODE REVIEW

File: gadget/params.c [hunk 1]
```diff
@@ -156,15 +156,16 @@ create_gadget_parameter_set()
     param_declare_double(ps, "ErrTolForceAcc", OPTIONAL, 0.005, "Force accuracy required from tree. Controls tree opening criteria. Lower values are more accurate.");
     param_declare_double(ps, "BHOpeningAngle", OPTIONAL, 0.175, "Barnes-Hut opening angle. Alternative purely geometric tree opening angle. Lower values are more accurate.");
     param_declare_int(ps, "TreeUseBH", OPTIONAL, 2, "If 1, use Barnes-Hut opening angle rather than the standard Gadget acceleration based opening angle. If 2, use BH criterion for the first timestep only, before we have relative accelerations.");
-    param_declare_double(ps, "Asmth", OPTIONAL, 1.25, "The scale of the short-range/long-range force split in units of FFT-mesh cells. Gadget-2 paper says larger values may be more accurate.");
+    param_declare_double(ps, "Asmth", OPTIONAL, 1.25, "The scale of the short-range/long-range force split in units of FFT-mesh cells."
+                                                      "Larger values suppresses grid anisotropy. ShortRangeForceWindowType = ercf supports any value. Exact only support 1.25. ");
```
Comment (sbird) [inline gadget/params.c]: Typo: ercf should be erfc

File: libgadget/shortrange-kernel.c [hunk 2]
```diff
@@ -0,0 +1,129 @@
+# x(in mesh units) w_pot_1d(x) w_force_1d(x) [erfc + other terms] w_pot_erf(x) w_force_erf(x) split=1.25
+0.000000000000000000e+00 1.000000000000000000e+00 1.000000000000000000e+00 1.000000000000000000e+00 1.000000000000000000e+00
+7.874015748031491790e-02 9.653015938297689891e-01 9.999769949380807876e-01 9.653130389985472570e-01 9.999765104804746452e-01
+1.574803149606298358e-01 9.306094442530644661e-01 9.998161995204686292e-01 9.306360761607166410e-01 9.998124189932160455e-01
+2.362204724409441570e-01 8.960005171189181494e-01 9.993814599961745593e-01 8.960393084328690527e-01 9.993687939840953938e-01
+3.149606299212620031e-01 8.615516177143599519e-01 9.985398797531048753e-01 8.615923428237660664e-01 9.985100199387387798e-01
+3.937007874015745479e-01 8.272900702259927685e-01 9.971610971054614136e-01 8.273637861144887973e-01 9.971053938911944581e-01
```
Comment (sbird) [inline libgadget/shortrange-kernel.c]: Do you need the extra columns? Not that they do any harm...

**Comment by sbird** [comment]
Apart from the obvious and easily fixable compile issue with the comment in shortrange-kernel.c, this looks good to me! What did the update to the kernel do?

**Comment by rainwoodman** [comment]
It is updated to be consistent with what the python script generates with 512 sample points  -- it was from an older version with 1000 sample points.

**Comment by sbird** [comment]
Merge it!

**Pull request merged**